### ISSUE TITLE
grpc-node tests broken on master

### ISSUE THREAD

**Issue by jtattermusch** [grpc/grpc#15470]
Failing in 100% cases

https://source.cloud.google.com/results/invocations/291b86bc-4f1e-4ed4-9dc2-a96ce6389458/targets/github%2Fgrpc%2Fgrpc-node_linux_dbg_native/tests

```
+ npm install --unsafe-perm
npm WARN deprecated gulp-util@3.0.8: gulp-util is deprecated - replace it, following the guidelines at https://medium.com/gulpjs/gulp-util-ca3b1f9f9ac5
npm WARN deprecated graceful-fs@3.0.11: please upgrade to graceful-fs 4 for compatibility with current and future versions of Node.js
npm WARN deprecated natives@1.1.3: This module relies on Node.js's internals and will break at some point. Do not use it, and update to graceful-fs@4.x.
npm WARN deprecated minimatch@2.0.10: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
npm WARN deprecated minimatch@0.2.14: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
npm WARN deprecated graceful-fs@1.2.3: please upgrade to graceful-fs 4 for compatibility with current and future versions of Node.js
npm notice created a lockfile as package-lock.json. You should commit this file.
added 954 packages from 822 contributors in 18.782s
[!] 12 vulnerabilities found [4680 packages audited]
    Severity: 5 Low | 4 High | 3 Critical
    Run `npm audit` for more detail
+ mkdir -p reports
+ export JOBS=8
+ export JUNIT_REPORT_STACK=1
+ uname
+ OS=Linux
+ FAILED=false
+ set +ex
Switching to node version 4
v4.9.1 is already installed.
/root/.nvm/versions/node/v4.9.1/lib/node_modules/npm/bin/npm-cli.js:79
      let notifier = require('update-notifier')({pkg})
      ^^^
SyntaxError: Block-scoped declarations (let, const, function, class) not yet supported outside strict mode
    at exports.runInThisContext (vm.js:53:16)
    at Module._compile (module.js:373:25)
    at Object.Module._extensions..js (module.js:416:10)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12)
    at Function.Module.runMain (module.js:441:10)
    at startup (node.js:140:18)
    at node.js:1043:3
nvm is not compatible with the npm config "prefix" option: currently set to ""
Run `nvm use --delete-prefix v4.9.1` to unset it.
/root/.nvm/versions/node/v4.9.1/lib/node_modules/npm/bin/npm-cli.js:79
      let notifier = require('update-notifier')({pkg})
      ^^^
SyntaxError: Block-scoped declarations (let, const, function, class) not yet supported outside strict mode
    at exports.runInThisContext (vm.js:53:16)
    at Module._compile (module.js:373:25)
    at Object.Module._extensions..js (module.js:416:10)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12)
    at Function.Module.runMain (module.js:441:10)
    at startup (node.js:140:18)
    at node.js:1043:3
nvm is not compatible with the npm config "prefix" option: currently set to ""
Run `nvm use --delete-prefix v4.9.1` to unset it.
+ export JUNIT_REPORT_PATH=reports/node4/
+ npm install -g node-gyp
./run-tests.sh: 62: ./run-tests.sh: npm: not found
Command exited with non-zero status 127
real 158.49
user 145.92
sys 58.13
```

**Comment by nicolasnoble** [comment]
This is an external breakage, it seems. Applying #15471 might have happened when the external package got fixed. Are we sure this is still going on otherwise ?

**Comment by nicolasnoble** [comment]
No it still seems to happen. That's really weird that #15471 might fix it.

**Comment by nicolasnoble** [comment]
The proper fix is probably rather in https://github.com/grpc/grpc-node/pull/356 actually.

**Comment by jtattermusch** [comment]
"Fixed" by merging #15471 to make things green again.
